File: examples/matlab/run_all.m
%RUN Runs all examples.

% Build interface
cd([pwd '/../../src/interface_matlab'])
pogs_setup -gpu % Alternatively `pogs_setup -gpu`
cd([pwd '/../../examples/matlab/'])
addpath([pwd '/../../src/interface_matlab'])


%% Run examples
fprintf('\nBasis Pursuit\n')
t = basis_pursuit(2000, 20000);
fprintf('Solver Time: %e sec\n', t)
%%
fprintf('\nEntropy\n')
t = entropy(200, 2000);
fprintf('Solver Time: %e sec\n', t)
%%
fprintf('\nHuber\n')
t = huber_fit(2000, 200);
fprintf('Solver Time: %e sec\n', t)
%%
fprintf('\nLasso\n')
t = lasso(100, 1000);
fprintf('Solver Time: %e sec\n', t)
%%
fprintf('\nLasso Path\n')
t = lasso_path(100, 1000);
fprintf('Solver Time: %e sec\n', t)
%%
fprintf('\nLinear Program in Equality Form\n')
t = lp_eq(200, 1000, struct('rho', 100, 'max_iter', 10000));
fprintf('Solver Time: %e sec\n', t)
%%
fprintf('\nLinear Program in Inequality Form\n')
t = lp_ineq(1000, 200, struct('rho', 1e-1, 'max_iter', 10000));
fprintf('Solver Time: %e sec\n', t)
%%
fprintf('\nLogistic Regression\n')
t = logistic(1000, 200);
fprintf('Solver Time: %e sec\n', t)
%%
fprintf('\nNon-Negative Least Squares\n')
t = nonneg_l2(1000, 200);
fprintf('Solver Time: %e sec\n', t)
%%
fprintf('\nPortfolio Optimization\n')
t = portfolio(200, 2000);
fprintf('Solver Time: %e sec\n', t)
%%
fprintf('\nPiecewise Linear Fitting\n')
t = pwl(2000, 200);
fprintf('Solver Time: %e sec\n', t)
%%
fprintf('\nSupport Vector Machine\n')
t = lasso(2000, 200);
fprintf('Solver Time: %e sec\n', t)

File: examples/py/run_all.py
from lasso import lasso
from lasso_path import lasso_path
from logistic import logistic
from lp_eq import lp_eq
from lp_ineq import lp_ineq
from nonneg_l2 import non_neg_l2
from svm import svm
import sys


def run_all(gpu=False, double_precision=False, m=1000, n=1000):
    print("\nLasso.")
    print("Solve time:\t{:.2e} seconds\n".format(lasso(m, n, gpu=gpu, double_precision=double_precision)))

    print("\nLasso Path.")
    print("Solve time:\t{:.2e} seconds\n".format(lasso_path(m, n, gpu=gpu, double_precision=double_precision)))

    print("\nLogistic Regression.")
    print("Solve time:\t{:.2e} seconds\n".format(logistic(m, n, gpu=gpu, double_precision=double_precision)))

    print("\nLinear Program in Equality Form.")
    print("Solve time:\t{:.2e} seconds\n".format(lp_eq(m, n, gpu=gpu, double_precision=double_precision)))

    print("\nLinear Program in Inequality Form.")
    print("Solve time:\t{:.2e} seconds\n".format(lp_ineq(m, n, gpu=gpu, double_precision=double_precision)))

    print("\nNon-Negative Least Squares.")
    print("Solve time:\t{:.2e} seconds\n".format(non_neg_l2(m, n, gpu=gpu, double_precision=double_precision)))

    print("\nSupport Vector Machine.")
    print("Solve time:\t{:.2e} seconds\n".format(svm(m, n, gpu=gpu, double_precision=double_precision)))


if __name__ == "__main__":
    gpu = True
    double = False
    m = 10000
    n = 10000
    if 'gpu' in sys.argv:
        gpu = True
    if 'double' in sys.argv:
        double = True
    run_all(gpu=gpu, double_precision=double, m=m, n=n)
